sync/src/orphan_block_pool.rs

```diff
@@ -72,6 +78,7 @@ impl InnerPool {
             }
         }
 
+        self.block_size -= removed.iter().map(|b| b.data().total_size()).sum::<usize>();
```

Comment:
Be more defensive here? Since if we mess up and the block size is wrapped around, we will keep in the special downloading mode for a long time.

```suggestion
        self.block_size = self.block_size.checked_sub(removed.iter().map(|b| b.data().total_size()).sum::<usize>()).unwrap_or_else(|| {
            // report error
            0
        });
```

Reply:
Both addition and subtraction were changed to checked, and then the error log was output, although I think there will be no problem.